### PR TITLE
Fix Arkansas capital gains calculations

### LINKED ISSUE #4385: Fix Arkansas capital gains calculations
In the 0.730.1 version, the formula for the new `ar_taxable_capital_gains` variable does not correspond to logic on the [AR1000D form](https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/2021_AR1000D_CapitalGains.pdf) in several ways.   I suggest reading the form and adding more unit tests of this variable.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4385

### CODE REVIEW

**Comment by PavelMakarchuk** [comment]
I missed basically the entire computation structure, now we modeled the [tax form](https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/2021_AR1000D_CapitalGains.pdf) line by line:
image

Now the cap of $10,000,000 in included, the 50% rate is applied to net capital gains and we also include a loss limit.

File: policyengine_us/tests/policy/baseline/gov/states/ar/tax/income/integration.yaml [hunk 1]
```diff
@@ -163,4 +132,4 @@
     ar_personal_credits: [261]
     ar_non_refundable_credits: 261
     ar_refundable_credits: 0
-    # ar_income_tax: 687.03 expected value from TAXSIM35
+    # ar_income_tax: 687.03 #expected value from TAXSIM35
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/ar/tax/income/integration.yaml:135]: should this be uncommented?

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 2]
```diff
@@ -10,12 +10,22 @@ class ar_taxable_capital_gains(Variable):
     defined_for = StateCode.AR
 
     def formula(person, period, parameters):
-        long_term_capital_gains = person("long_term_capital_gains", period)
-        short_term_capital_gains = person("short_term_capital_gains", period)
+        net_lt_capital_gains = person("long_term_capital_gains", period)
+        net_st_capital_gains = person("short_term_capital_gains", period)
+        # Can be capital loss
+        net_cap_gain = net_lt_capital_gains - net_st_capital_gains
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:16]: what line does this correspond to? i don't see where short-term net gains would be subtracted from long-term net gains

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 3]
```diff
@@ -10,12 +10,22 @@ class ar_taxable_capital_gains(Variable):
...
-            1 - p.exempt_rate
+        capped_net_cap_gain = min_(net_cap_gain, p.exempt.cap)
+        taxable_capped_net_cap_gain = capped_net_cap_gain * (1 - p.exempt.rate)
+        st_capital_gain = max_(0, net_st_capital_gains)
+        taxable_capital_gain = st_capital_gain + taxable_capped_net_cap_gain
+
+        capital_loss = taxable_capital_gain < 0
+        # Taxable capital loss is capped separately
+        filing_status = person.tax_unit("filing_status", period)
+        capped_capital_loss = max_(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:28]: if cap shouldn't it be min?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py (hunk 2 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:16]: Line 7a - the naming net lt gains was wrong - fixed

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py (hunk 3 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:28]: It caps negative gains to 1,500, I have added a multiplication by -1 to more accurately represent that case:
```
        capital_loss = taxable_capital_gain < 0
        # Taxable capital loss is capped separately
        filing_status = person.tax_unit("filing_status", period)
        capped_capital_loss = min_(
            (taxable_capital_gain * (-1)), p.loss_cap[filing_status]
        )
        return where(capital_loss, capped_capital_loss, taxable_capital_gain)

```

File: policyengine_us/tests/policy/baseline/gov/states/ar/tax/income/integration.yaml (hunk 1 above)
Comment (PavelMakarchuk) [reply policyengine_us/tests/policy/baseline/gov/states/ar/tax/income/integration.yaml:135]: no, this is from a previous integration test and we are still working out which model is correct, this change does not relate to this PR

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 4]
```diff
@@ -10,12 +10,22 @@ class ar_taxable_capital_gains(Variable):
...
             period
         ).gov.states.ar.tax.income.gross_income.capital_gains
-        taxable_long_term_capital_gains = long_term_capital_gains * (
-            1 - p.exempt_rate
+        capped_net_cap_gain = min_(net_cap_gain, p.exempt.cap)
+        taxable_capped_net_cap_gain = capped_net_cap_gain * (1 - p.exempt.rate)
+        st_capital_gain = max_(0, net_st_capital_gains)
+        taxable_capital_gain = st_capital_gain + taxable_capped_net_cap_gain
+
+        capital_loss = taxable_capital_gain < 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:25]:
```suggestion
        taxable_capital_loss = -taxable_capital_gain
        has_taxable_capital_loss = taxable_capital_loss > 0
```

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 5]
```diff
@@ -10,12 +10,22 @@ class ar_taxable_capital_gains(Variable):
...
+        # Taxable capital loss is capped separately
+        filing_status = person.tax_unit("filing_status", period)
+        capped_capital_loss = min_(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:28]:
```suggestion
        loss_cap = p.loss_cap[filing_status]
        capped_capital_loss = min_(taxable_capital_loss, loss_cap)
```

File: policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py [hunk 6]
```diff
@@ -10,12 +10,22 @@ class ar_taxable_capital_gains(Variable):
...
+            (taxable_capital_gain * (-1)), p.loss_cap[filing_status]
         )
-        return short_term_capital_gains + taxable_long_term_capital_gains
+        return where(capital_loss, capped_capital_loss, taxable_capital_gain)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_taxable_capital_gains.py:31]:
```suggestion
        return where(has_taxable_capital_loss, capped_capital_loss, taxable_capital_gain)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
not sure if these are the perfect names but something like this
